evalai/challenges.py

```diff
@@ -67,10 +69,54 @@ def list_future_challenges():
     get_future_challenge_list()
 
 
+@click.group(invoke_without_command=True)
+@click.pass_context
+@click.option('-c', '--challenge-id', type=int,
+              help="Challenge ID for viewing its phases.")
+@click.option('-p', '--phase-id', type=int,
+              help="Phase ID for showing the phase details")
+def phases(ctx, challenge_id, phase_id):
+    """
+    Displays phase and phase related details.
+    Invoked by running `evalai challenges phases`
+    """
+    if ctx.invoked_subcommand != "list":
+        if challenge_id is None or phase_id is None:
```

Comment:
@varunagrawal Also, since the list phases is a child of this command, if the `required=True` is set then while running the child command the arguments of the parent have to be set.

Reply:
I don't quite understand what is the point of making something an option if it is required. The word option means that it is optionally but clearly the use case indicates that it is not. Using `required=True` is an ugly hack that can be very confusing to anyone who reads this function.